ctats/counter.go

```diff
@@ -31,14 +32,21 @@ func counterFromCtx(
 func getOrCreateCounter(
 	ctx context.Context,
 	id string,
-) (metric.Float64UpDownCounter, error) {
+) (adder, error) {
 	id = formatID(id)
 
-	ctr := counterFromCtx(ctx, id)
+	b, ctr := counterFromCtx(ctx, id)
 	if ctr != nil {
 		return ctr, nil
 	}
 
+	if b.initializedToNoop {
```

Comment:
🤔 The current implementation is good to go, but maybe in the future we could think about having an interface around `bus` so that there's no need to check for flags on it. Callers could just do `bus.NewCounter()` (or `bus.GetOrRegisterCounter`?) and go from there

Reply:
Yup, I like that idea.